client/src/main/java/io/pravega/client/tables/impl/KeyUpdateListener.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.tables.impl;
+
+import com.google.common.annotations.Beta;
+import io.pravega.client.tables.TableEntry;
+import io.pravega.client.tables.TableKey;
+import java.util.function.Consumer;
+
+/**
+ * Defines a Listener for Key Updates.
+ *
+ * Notes:
+ * * All registered callbacks will be invoked asynchronously on an {@link java.util.concurrent.Executor} that is either
+ * provided during construction of this instance or on another {@link java.util.concurrent.Executor}. Due to the asynchronous
+ * nature of such callbacks, it is not guaranteed that they will be invoked in order, so it is the responsibility of the
```

Comment:
If we wait on the callback to complete before invoking the next one, that will increase the likelihood of a backup to occur either on the client, server or somewhere in between. I want to avoid that situation.

Reply:
The client's pulling this, much like a normal read right? what's wrong with it being behind?